internal/analytics/client.go

```diff
@@ -7,65 +7,64 @@ import (
 	"io"
 	"net/http"
 	"sync"
+	"time"
 
 	"github.com/stretchr/testify/mock"
 )
 
+type TrackerFn func(accessToken string, baseURI string, optOut bool) Tracker
+
+type ClientFn struct {
+	ID string
+}
+
+func (fn ClientFn) Tracker(version string) TrackerFn {
+	return func(accessToken string, baseURI string, optOut bool) Tracker {
+		if optOut {
```

Comment:
This encapsulates the opt-out flag so we don't need to check it when calling the send methods. If the user opts-out, all the send methods are no-ops.